src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs

```diff
@@ -294,6 +321,90 @@ private async Task WriteChange(ResourceViewModel resourceViewModel, ObjectChange
             .ConfigureAwait(false);
     }
 
+    // Check a service endpoint
+    private async Task<bool> CheckServiceEndpointReady(string proxyUrlString, string errorContent)
+    {
+        // Explicitly use a new HttpClient for each request to avoid re-using the same socket connection
+        using HttpClient client = new();
```

Comment:
I am a little worried about this, and I wonder if we should use a singleton http client instance configured like this:

```C#
var handler = new SocketsHttpHandler
{
    PooledConnectionLifetime = TimeSpan.Zero
};

var client = new HttpClient(handler);
```

@JamesNK any thoughts?

cc @MihaZupan

Reply:
Just tested and everything still works with a singleton; seems there's no reason not to do this

Reply:
Aren't they equivalent from an unmanaged resource perspective? (TCP sockets) They're both creating a new socket connection for each request.

But a singleton SocketsHttpHandler is probably kinder on allocations on .NET (reuse of all of SocketsHttpHandler's infrastructure instead of starting from a blank slate) so there is an improvement there.

Reply:
Right there’s lots of state associated with the handler that we can avoid tearing down